### ISSUE TITLE
Update CyclonePlotter for offline/HPC usage

### ISSUE THREAD

**Issue by j-opatz** [dtcenter/METplus#933]
## Describe the Problem ##
In a recent test of a use case involving CyclonePlotter on Hera, the script error'd out. After looking into the cause, it was determined that cartopy, one of the packages used in CyclonePlotter, was attempting to download a zip file of map information. While this could be accomplished on NCAR development servers with internet connection, it's a non-starter on the rdhpcs with NOAA.

### Expected Behavior ###
CyclonePlotter needs to be able to successfully complete in offline modes. All packages necessary for running should be locally found, with no online calls needed.

### Environment ###
Describe your runtime environment:
*1. Machine: Hera
*2. OS: Linux
*3. Versions: METplus 4.0.0, MET 10.0.0
### To Reproduce ###
Describe the steps to reproduce the behavior:
*1. In Hera, run use case CyclonePlotter_fcstGFS_obsGFS_OPC
*2. Error will occur during CyclonePlotter section (posted below)
*3. /contrib/anaconda/anaconda3/latest/lib/python3.7/site-packages/cartopy/io/__init__.py:260: DownloadWarning: Downloading: http://naciscdn.org/naturalearth/110m/physical/ne_110m_land.zip
  warnings.warn('Downloading: {}'.format(url), DownloadWarning)
05/17 18:33:37.504 metplus.CyclonePlotter (met_util.py:186) ERROR: Fatal error occurred
Traceback (most recent call last):
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 928, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

*Post relevant sample data following these instructions:*
trak.gfso.atcf_gen.glbl.2020 (included in METplus 4.0.0 tarball)

### Relevant Deadlines ###
Next bugfix

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to relevant bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
- [ ] Close this issue.

**Comment by bikegeek** [comment]
A possible work around is to download the zip file and save it to a specified directory, then define an environment variable that indicates where downloaded file is located.  Then add something like this in the cyclone plotter code:

import cartopy
cartopy.config['data_dir'] = os.getenv('CARTOPY_DIR', cartopy.config.get('data_dir'))

as specified in the issues for Cartopy:
https://github.com/SciTools/cartopy/issues/1325

**Note,** this _might not_ work.

**Comment by j-opatz** [comment]
As of today, the patch method put together by @bikegeek in the CyclonePlotter script, along with @jprestop 's downloading of the zip file to Hera and setting the CARTOPY_DIR environment to the zip file area, was successful.
I need to quickly test (on kiowa) this same branch to verify the updated CyclonePlotter script still works if the CARTOPY_DIR variable isn't set (as indicated in @bikegeek last code commit). Once successful, I'll update documentation in the Use Cases and script, and create a PR.

Update: the branch works on Kiowa so the non-setting method also works as normal. Documentation updates are underway, and a PR should be set soon.

**Comment by j-opatz** [comment]
This awaits testing by Guang Ping, who has been on vacation during the available testing time. As the beta is set to release at EOM, I'm moving this issue to the next beta

Update: as a beta's purpose is to provide a testing area for users to try new updates/fixes, this issue is being added back to beta1 and should be completed before the EOM.
